ultratrace2/model/project.py

```diff
@@ -1,23 +1,40 @@
 import os
+import pickle
 
 from .trace import TraceList
 from .files.bundle import FileBundleList
 
 
 class Project:
     def __init__(self, path: str):
+        """
+        Internal function: to construct a Project, either call via ::initialize_from_path()
+        or ::load().
+        """
         if not os.path.exists(path):
             raise ValueError(f"cannot initialize project at {path}")
         self.root_path = os.path.realpath(os.path.abspath(path))  # absolute path
         self.traces = TraceList()
-        self.files = FileBundleList(self.root_path)
+        self.files = FileBundleList()
 
     def save(self):
         raise NotImplementedError()
 
     @classmethod
-    def load(cls):
-        raise NotImplementedError()
+    def load(cls, project_path) -> "Project":
+        with open(project_path, "rb") as fp:
+            project = pickle.load(fp)
+            assert isinstance(project, Project)
+            return project
+
+    @classmethod
+    def initialize_from_path(cls, data_path) -> "Project":
+        project = Project(data_path)
```

Comment:
In order to avoid half-initialized objects, add take an instance of `FileBundleList` in `__init__`, then in this alternate constructor, create a new instance of `FileBundleList`, call `scan_dir`, then pass it to the `Project` constructor.

Also, since this is a `classmethod`, you can use `cls` to invoke `__init__`:
```suggestion
        project = cls(data_path)
```

Reply:
Specifically in reference to this comment, do you mind taking a look at https://github.com/SwatPhonLab/UltraTrace/commit/7c46ccb94bf866a69bef140d71a2d4a29de64b24 ?